contracts/plugins/assets/stargate/StargatePoolFiatCollateral.sol

```diff
@@ -0,0 +1,121 @@
+        view
+        virtual
+        override
+        returns (
+            uint192 low,
+            uint192 high,
+            uint192 pegPrice
+        )
+    {
+        pegPrice = chainlinkFeed.price(oracleTimeout); // {target/ref}
+
+        // Assumption: {UoA/target} = 1; target is same as UoA
+        // {UoA/tok} = {UoA/target} * {target/ref} * {ref/tok}
+        uint192 p = pegPrice.mul(refPerTok());
+
+        // {UoA/tok} = {UoA/tok} * {1}
+        uint192 delta = p.mul(oracleError);
+
+        low = p - delta;
+        high = p + delta;
+    }
+
+    /// Should not revert
+    /// Refresh exchange rates and update default status.
+    /// @dev Should not need to override: can handle collateral with variable refPerTok()
+    function refresh() public virtual override {
+        if (alreadyDefaulted()) return;
+
+        CollateralStatus oldStatus = status();
+
+        // Check for hard default
+
+        uint192 referencePrice = refPerTok();
+        uint192 lastReferencePrice_ = lastReferencePrice;
+
+        // uint192(<) is equivalent to Fix.lt
+        if (referencePrice < lastReferencePrice_) {
+            markStatus(CollateralStatus.DISABLED);
+        } else if (referencePrice > lastReferencePrice_) {
+            lastReferencePrice = referencePrice;
+        }
+
+        // Check for soft default + save prices
+        try this.tryPrice() returns (uint192 low, uint192 high, uint192 pegPrice) {
+            // {UoA/tok}, {UoA/tok}, {target/ref}
+            // (0, 0) is a valid price; (0, FIX_MAX) is unpriced
+
+            // Save prices if priced
+            if (high < FIX_MAX) {
+                savedLowPrice = low;
+                savedHighPrice = high;
+                lastSave = uint48(block.timestamp);
+            } else {
+                // must be unpriced
+                assert(low == 0);
+            }
+
+            // If the price is below the default-threshold price, default eventually
+            // uint192(+/-) is the same as Fix.plus/minus
+            if (pegPrice < pegBottom || pegPrice > pegTop || low == 0) {
+                markStatus(CollateralStatus.IFFY);
+            } else {
+                markStatus(CollateralStatus.SOUND);
+            }
+        } catch (bytes memory errData) {
+            // see: docs/solidity-style.md#Catching-Empty-Data
+            if (errData.length == 0) revert(); // solhint-disable-line reason-string
+            markStatus(CollateralStatus.IFFY);
+        }
+
+        CollateralStatus newStatus = status();
+        if (oldStatus != newStatus) {
+            emit CollateralStatusChanged(oldStatus, newStatus);
+        }
+    }
+
+    /// @return _rate {ref/tok} Quantity of whole reference units per whole collateral tokens
+    function refPerTok() public view virtual override returns (uint192 _rate) {
+        IStargatePoolWrapper poolWrapper = IStargatePoolWrapper(address(erc20));
+        IStargatePool pool = IStargatePool(address(poolWrapper.pool()));
```

Comment:
Since it's immutable we should probably store `poolWrapper.pool()` and avoid re-lookup